File: data/EXP01/go/2348/2348.output_3.go
func zeroFilledSubarray(nums []int) int64 {
    count := int64(0)
    zeroCount := 0
    zeroMap := make(map[int]int64)

    for _, num := range nums {
        if num == 0 {
            zeroCount++
        } else {
            count += int64(zeroCount*(zeroCount+1)/2) // Calculate the number of subarrays formed by zeros
            zeroCount = 0
        }
        count += zeroMap[zeroCount] // Add the count of subarrays with the current zero count
        zeroMap[zeroCount]++ // Increment the count of subarrays with the current zero count
    }

    count += int64(zeroCount*(zeroCount+1)/2) // Calculate the number of subarrays formed by zeros at the end
    return count
}

File: data/EXP01/python/2348/2348.output_3.py
# @lc app=leetcode id=2348 lang=python3
from typing import List

class Solution:
    def zeroFilledSubarray(self, nums: List[int]) -> int:
        count = 0
        left = 0
        zeros = 0
        
        for right in range(len(nums)):
            if nums[right] == 0:
                zeros += 1
                
            if right - left + 1 > zeros:
                count += right - left + 1 - zeros
                
            while zeros > 0:
                if nums[left] == 0:
                    zeros -= 1
                left += 1
                    
        return count